Sources/AblyAssetTrackingSubscriber/WorkerQueue/SubscriberWorkerQueueProperties.swift

```diff
@@ -3,4 +3,185 @@ import Foundation
+        // because it's possible to have presence entry events without subsequent leave events.
+        // Therefore, from the perspective of a user consuming events from publisherPresenceStateFlow, what matters
+        // is what's computed for isPublisherVisible (not the simple isAPublisherPresent).
+        var isPublisherVisible = false
+        if case .online = lastConnectionStateChange.state {
+            if isAPublisherPresent {
+                isPublisherVisible = true
+            }
+        }
+        
+        if lastEmittedValueOfIsPublisherVisible != isPublisherVisible {
+            lastEmittedValueOfIsPublisherVisible = isPublisherVisible
+            notifyPublisherPresenceUpdated(isPublisherPresent: isPublisherVisible)
+        }
+        
+        notifyResolutionsChanged(resolutions: pendingPublisherResolutions.drain())
+    }
+    
+    mutating func notifyEnhancedLocationUpdated(locationUpdate: LocationUpdate) {
+        enhancedLocation = locationUpdate
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didUpdateEnhancedLocation: locationUpdate)
+    }
+    
+    mutating func notifyRawLocationUpdated(locationUpdate: LocationUpdate) {
+        rawLocation = locationUpdate
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didUpdateRawLocation: locationUpdate)
+    }
+    
+    mutating func notifyPublisherPresenceUpdated(isPublisherPresent: Bool) {
+        publisherPresence = isPublisherPresent
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didUpdatePublisherPresence: publisherPresence)
+    }
+    
+    mutating func notifyTrackableStateUpdated(trackableState: ConnectionState) {
+        self.trackableState = trackableState
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didChangeAssetConnectionStatus: trackableState)
+    }
+    
+    mutating func notifyDidFailWithError(error: ErrorInformation) {
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didFailWithError: error)
+    }
+    
+    mutating func notifyResolutionsChanged(resolutions: [Resolution]) {
+        if !resolutions.isEmpty {
+            for resolution in resolutions {
+                self.resolution = resolution
+                self.nextLocationUpdateInterval = resolution.desiredInterval
+                guard let subscriber = subscriber
+                else { return }
+                delegate?.subscriber(sender: subscriber, didUpdateResolution: resolution)
+                delegate?.subscriber(sender: subscriber, didUpdateDesiredInterval: resolution.desiredInterval)
+            }
+        }
+    }
+}
+
+fileprivate class PendingResolutions {
```

Comment:
Should this not be a struct, to preserve the value semantics of `SubscriberWorkerQueueProperties`?

Reply:
I'm not sure it would actually cause problems in our case, but sure, it can and probably should actually be just a struct. a1f35512729b12de1185bb4762a56d31488296c1